src/components/script/ScriptEventForm.js

```diff
@@ -53,6 +56,71 @@ class ScriptEventForm extends Component {
 
       return [].concat(description, eventCommands, usedVariables, usedActors);
     }
+    if ((command === EVENT_ENGINE_FIELD_UPDATE || command === EVENT_ENGINE_FIELD_STORE) && value.engineFieldKey) {
+      const engineField = engineFields.find((e) => e.key === value.engineFieldKey);
+      if (engineField) {
+        const fieldType = engineField.type || "number";
+        if (command === EVENT_ENGINE_FIELD_UPDATE) {
+          if (is16BitCType(engineField.cType)) {
+            return [].concat(
+              eventCommands,
+              {
+                key: "value",
+                type: "union",
+                checkboxLabel: l10n(engineField.label),
+                types: [fieldType, "variablePair"],
+                defaultType: fieldType,
+                min: clampToCType(engineField.min || -Infinity, engineField.cType),
+                max: clampToCType(engineField.max || Infinity, engineField.cType),
+                options: engineField.options || [],
+                defaultValue: {
+                  [fieldType]: engineField.defaultValue || 0,
+                  variable: "0:0",
+                },
+              }
+            );
+          } 
+          return [].concat(eventCommands, {
+            key: "value",
+            type: "union",
+            checkboxLabel: l10n(engineField.label),
+            types: [fieldType, "variable"],
+            defaultType: fieldType,
+            min: clampToCType(engineField.min || -Infinity, engineField.cType),
+            max: clampToCType(engineField.max || Infinity, engineField.cType),
+            options: engineField.options || [],
+            defaultValue: {
+              [fieldType]: engineField.defaultValue || 0,
+              variable: "0",
+            },
+          });
+        }
+        if (command === EVENT_ENGINE_FIELD_STORE) {
+          if (is16BitCType(engineField.cType)) {
+            return [].concat(
+              eventCommands,
+              {
+                key: "value",
```

Comment:
Loading those dynamically will make more difficult (maybe even not possible) to identify the fields as variables inside a Custom Event. Because the `type` field won't be available neither in the event instance nor in the event declaration :/